legion@21.03.0 +rocm: cmake: could not find requested file: /opt/rocm-5.3.0/cmake/FindHIP.cmake

### Steps to reproduce the issue

`legion@21.03.0 +rocm` is failing to build on OLCF Frontier:
* PrgEnv-gnu, gcc/11.2.0
* rocm/5.3.0

Spack environment: [spack.yaml.txt](https://github.com/spack/spack/files/11337002/spack.yaml.txt)

<details><summary>Concretization</summary>
<pre>
 -   yrf2tmj  legion@21.03.0%gcc@11.2.0~bindings~bounds_checks~cuda~cuda_hijack~cuda_unsupported_compiler~enable_tls~fortran~gasnet_debug~hdf5~hip_hijack~hwloc~ipo~kokkos+libdl~openmp~papi~privilege_checks~python~redop_complex+rocm~shared~spy+zlib amdgpu_target=gfx90a build_system=cmake build_type=RelWithDebInfo conduit=none cuda_arch=70 cxxstd=11 gasnet_root=none hip_target=ROCM max_dims=3 max_fields=512 network=none output_level=warning arch=linux-sles15-zen3
[+]  btskcfz      ^cmake@3.25.2%gcc@11.2.0~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-sles15-zen3
[+]  igbrz2c          ^ncurses@6.3%gcc@11.2.0~symlinks+termlib abi=none build_system=autotools arch=linux-sles15-zen3
[+]  savxweu              ^pkgconf@1.8.0%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen3
[+]  2r77ale          ^openssl@1.1.1s%gcc@11.2.0~docs~shared build_system=generic certs=mozilla arch=linux-sles15-zen3
[+]  grqzkfl              ^ca-certificates-mozilla@2023-01-10%gcc@11.2.0 build_system=generic arch=linux-sles15-zen3
[+]  g7oioq5              ^perl@5.36.0%gcc@11.2.0+cpanm+open+shared+threads build_system=generic arch=linux-sles15-zen3
[+]  gqdvawb                  ^berkeley-db@18.1.40%gcc@11.2.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-sles15-zen3
[+]  6pg5v4t                  ^bzip2@1.0.8%gcc@11.2.0~debug~pic+shared build_system=generic arch=linux-sles15-zen3
[+]  2rhfndx                      ^diffutils@3.8%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen3
[+]  hr2sqld                          ^libiconv@1.17%gcc@11.2.0 build_system=autotools libs=shared,static arch=linux-sles15-zen3
[+]  gscnwqx                  ^gdbm@1.23%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen3
[+]  n7oc7lg                      ^readline@8.2%gcc@11.2.0 build_system=autotools patches=bbf97f1 arch=linux-sles15-zen3
[+]  zmrrfsf      ^hip@5.3.0%gcc@11.2.0~cuda~ipo+rocm build_system=cmake build_type=Release patches=ca523f1 arch=linux-sles15-zen3
[+]  kzscxzp      ^hsa-rocr-dev@5.3.0%gcc@11.2.0+image~ipo+shared build_system=cmake build_type=Release patches=71e6851 arch=linux-sles15-zen3
[+]  5wkmp7b      ^kokkos@3.7.01%gcc@11.2.0~aggressive_vectorization~compiler_warnings~cuda~cuda_constexpr~cuda_lambda~cuda_ldg_intrinsic~cuda_relocatable_device_code~cuda_uvm~debug~debug_bounds_check~debug_dualview_modify_check~deprecated_code~examples~explicit_instantiation~hpx~hpx_async_dispatch~hwloc~ipo~memkind~numactl~openmp~openmptarget~pic+profiling~profiling_load_print~pthread~qthread+rocm+serial+shared~sycl~tests~tuning~wrapper amdgpu_target=gfx90a build_system=cmake build_type=RelWithDebInfo intel_gpu_arch=none std=14 arch=linux-sles15-zen3
[+]  jey42pp      ^llvm-amdgpu@5.3.0%gcc@11.2.0~ipo~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_system=cmake build_type=Release patches=a08bbe1 arch=linux-sles15-zen3
[+]  d35p7mx      ^zlib@1.2.13%gcc@11.2.0+optimize+pic+shared build_system=makefile arch=linux-sles15-zen3
</pre></details>


Error:
```
$> spack -e . install
...
==> Installing legion-21.03.0-yrf2tmjktcymodyd56v3skmxtzu57u3a
==> No binary for legion-21.03.0-yrf2tmjktcymodyd56v3skmxtzu57u3a found: installing from source
==> Using cached archive: /mnt/bb/lpeyrala/PrgEnv-gnu/spack/var/spack/cache/_source-cache/git//StanfordLegion/legion.git/legion-21.03.0.tar.gz
==> No patches needed for legion
==> Warning: Fetching from mirror without a checksum!
  This package is normally checked out from a version control system, but it has been archived on a spack mirror.  This means we cannot know a checksum for the tarball in advance. Be sure that your connection to this mirror is secure!
==> legion: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/mnt/bb/lpeyrala/PrgEnv-gnu/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholde/linux-sles15-zen3/gcc-11.2.0/cmake-3.25.2-btskcfzaljlsbup6mlvd6zrdxbhxteqr/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/mnt/bb/lpeyrala/PrgEnv-gnu/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholde/linux-sles15-zen3/gcc-11.2.0/legion-21.03.0-yrf2tmjktcymodyd56v3skmxtzu57u3a' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/mnt/bb/lpeyrala/PrgEnv-gnu/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholde/linux-sles15-zen3/gcc-11.2.0/legion-21.03.0-yrf2tmjktcymodyd56v3skmxtzu57u3a/lib;/mnt/bb/lpeyrala/PrgEnv-gnu/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholde/linux-sles15-zen3/gcc-11.2.0/legion-21.03.0-yrf2tmjktcymodyd56v3skmxtzu57u3a/lib64;/opt/rocm-5.3.0/lib;/opt/rocm-5.3.0/llvm/lib;/mnt/bb/lpeyrala/PrgEnv-gnu/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholde/linux-sles15-zen3/gcc-11.2.0/zlib-1.2.13-d35p7mxmwabhpoqwah7g3vlgawooct7s/lib;/mnt/bb/lpeyrala/PrgEnv-gnu/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholde/linux-sles15-zen3/gcc-11.2.0/kokkos-3.7.01-5wkmp7bxzp2poewaqhb3sd7tz6pkkw2p/lib64;/opt/cray/pe/gcc/11.2.0/snos' '-DCMAKE_PREFIX_PATH:STRING=/mnt/bb/lpeyrala/PrgEnv-gnu/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholde/linux-sles15-zen3/gcc-11.2.0/kokkos-3.7.01-5wkmp7bxzp2poewaqhb3sd7tz6pkkw2p;/mnt/bb/lpeyrala/PrgEnv-gnu/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholde/linux-sles15-zen3/gcc-11.2.0/cmake-3.25.2-btskcfzaljlsbup6mlvd6zrdxbhxteqr;/mnt/bb/lpeyrala/PrgEnv-gnu/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholde/linux-sles15-zen3/gcc-11.2.0/zlib-1.2.13-d35p7mxmwabhpoqwah7g3vlgawooct7s;/opt/rocm-5.3.0/llvm;/opt/rocm-5.3.0/;/opt/rocm-5.3.0' '-DCMAKE_CXX_STANDARD:STRING=11' '-DLegion_EMBED_GASNet=OFF' '-DBUILD_SHARED_LIBS=OFF' '-DLegion_USE_HIP=ON' '-DLegion_GPU_REDUCTIONS=ON' '-DLegion_HIP_TARGET:STRING=ROCM' '-DLegion_HIP_ARCH:STRING=gfx90a' '-DLegion_HIJACK_HIP:BOOL=OFF' '-DLegion_USE_LIBDL=ON' '-DLegion_USE_ZLIB=ON' '-DLegion_MAX_DIM=3' '-DLegion_MAX_FIELDS=512' '-DBUILD_MARCH:STRING=' '/tmp/lpeyrala/spack-stage/spack-stage-legion-21.03.0-yrf2tmjktcymodyd56v3skmxtzu57u3a/spack-src'

2 errors found in build log:
     21    -- Performing Test COMPILER_SUPPORTS_MABI_ALTIVEC
     22    -- Performing Test COMPILER_SUPPORTS_MABI_ALTIVEC - Failed
     23    -- Performing Test COMPILER_SUPPORTS_MVSX
     24    -- Performing Test COMPILER_SUPPORTS_MVSX - Failed
     25    -- Performing Test COMPILER_SUPPORTS_DEFCHECK
     26    -- Performing Test COMPILER_SUPPORTS_DEFCHECK - Failed
  >> 27    CMake Error at cmake/FindHIP.cmake:25 (include):
     28      include could not find requested file:
     29    
     30        /opt/rocm-5.3.0/cmake/FindHIP.cmake
     31    Call Stack (most recent call first):
     32      CMakeLists.txt:317 (find_package)
     33    

     ...

     39    -- Looking for pthread_create in pthreads - not found
     40    -- Looking for pthread_create in pthread
     41    -- Looking for pthread_create in pthread - found
     42    -- Found Threads: TRUE
     43    -- Performing Test COMPILER_SUPORTS_OVERLOADED_VIRTUAL_WARNING
     44    -- Performing Test COMPILER_SUPORTS_OVERLOADED_VIRTUAL_WARNING - Success
  >> 45    CMake Error at cmake/FindHIP.cmake:33 (HIP_GET_SOURCES_AND_OPTIONS):
     46      Unknown CMake command "HIP_GET_SOURCES_AND_OPTIONS".
     47    Call Stack (most recent call first):
     48      cmake/FindHIP.cmake:47 (hip_compile_base)
     49      runtime/CMakeLists.txt:791 (hip_compile)
```

### Error message

<details><summary>Error message</summary>

<pre>
see ab ove
</pre></details>


### Information on your system

* **Spack:** 0.20.0.dev0 (4e811bb81e74b6bd4aa689ee7b25693495b3ae31)
* **Python:** 3.6.15
* **Platform:** linux-sles15-zen3
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/11337018/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/11337019/spack-build-out.txt)
[CMakeError.log](https://github.com/spack/spack/files/11337020/CMakeError.log)
[CMakeOutput.log](https://github.com/spack/spack/files/11337021/CMakeOutput.log)

@pmccormick @streichler @wspear

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

The rocm installs on frontier do not place the cmake directory in the top level directory. Instead it is found in the hip or lib sub-directory, like this:
./rocm-5.3.0/hip/cmake/FindHIP.cmake
./rocm-5.3.0/lib/cmake/hip/FindHIP.cmake

Adding "/hip" to the HIP_PATH variable in the legion package.py on line 273 like this:
`            build_env.set("HIP_PATH", spec["hip"].prefix+"/hip")`

bypasses the error. (A proper fix needs logic to detect the location of FindHIP.cmake) However, there is then a build error like this:
```

==> Installing legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup
==> No binary for legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup found: installing from source
==> Using cached archive: /lustre/orion/csc439/scratch/wspear/crusher/23.02/spack/var/spack/cache/_source-cache/git//StanfordLegion/legion.git/legion-21.03.0.tar.gz
==> Warning: Fetching from mirror without a checksum!
  This package is normally checked out from a version control system, but it has been archived on a spack mirror.  This means we cannot know a checksum for the tarball in advance. Be sure that your connection to this mirror is secure!
==> No patches needed for legion
==> legion: Executing phase: 'cmake'
==> legion: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

5 errors found in build log:
     168    cd /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime/CMakeFiles/hip_compile.dir && /lustre/orion/csc439/scratch/wspear/crusher/23.02/spack/opt/spack/linux-sles15-zen3/gcc-11.2.0/cmake-3.25.2-bts
            kcfzaljlsbup6mlvd6zrdxbhxteqr/bin/cmake -E make_directory /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime/CMakeFiles/hip_compile.dir//.
     169    cd /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime/CMakeFiles/hip_compile.dir && /lustre/orion/csc439/scratch/wspear/crusher/23.02/spack/opt/spack/linux-sles15-zen3/gcc-11.2.0/cmake-3.25.2-bts
            kcfzaljlsbup6mlvd6zrdxbhxteqr/bin/cmake -D verbose:BOOL=1 -D build_configuration:STRING=RELWITHDEBINFO -D generated_file:STRING=/tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime/CMakeFiles/hip_c
            ompile.dir//./hip_compile_generated_legion_redop.cpp.o -P /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime/CMakeFiles/hip_compile.dir//hip_compile_generated_legion_redop.cpp.o.cmake
     170    -- Removing /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime/CMakeFiles/hip_compile.dir//./hip_compile_generated_legion_redop.cpp.o
     171    /lustre/orion/csc439/scratch/wspear/crusher/23.02/spack/opt/spack/linux-sles15-zen3/gcc-11.2.0/cmake-3.25.2-btskcfzaljlsbup6mlvd6zrdxbhxteqr/bin/cmake -E remove /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build
            -wi3oukk/runtime/CMakeFiles/hip_compile.dir//./hip_compile_generated_legion_redop.cpp.o
     172    -- Generating dependency file: /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime/CMakeFiles/hip_compile.dir//hip_compile_generated_legion_redop.cpp.o.depend.pre
     173    /opt/rocm-5.4.0/bin/hipcc -M /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime/legion_redop.cpp -o /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build
            -wi3oukk/runtime/CMakeFiles/hip_compile.dir//hip_compile_generated_legion_redop.cpp.o.depend.pre
  >> 174    /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime/legion_redop.cpp:19:10: fatal error: 'realm/hip/hiphijack_api.h' file not found
     175    #include "realm/hip/hiphijack_api.h"
     176             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
     177    1 error generated when compiling for host.
  >> 178    CMake Error at hip_compile_generated_legion_redop.cpp.o.cmake:146 (message):
     179      Error generating
     180      /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime/CMakeFiles/hip_compile.dir//./hip_compile_generated_legion_redop.cpp.o
     181    
     182    
  >> 183    make[2]: *** [runtime/CMakeFiles/hip_compile.dir/build.make:80: runtime/CMakeFiles/hip_compile.dir/hip_compile_generated_legion_redop.cpp.o] Error 1
     184    make[2]: Leaving directory '/tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk'
  >> 185    make[1]: *** [CMakeFiles/Makefile2:225: runtime/CMakeFiles/hip_compile.dir/all] Error 2
     186    make[1]: *** Waiting for unfinished jobs....
     187    [ 30%] Building CXX object runtime/CMakeFiles/RealmRuntime.dir/realm/mem_impl.cc.o
     188    cd /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime && /lustre/orion/csc439/scratch/wspear/crusher/23.02/spack/lib/spack/env/gcc/g++  -I/opt/rocm-5.4.0/include -I/opt/rocm-5.4.0/../include -I/t
            mp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-src/runtime -I/tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime -march=native -D__HIP_PLATFORM_HCC__ -O2 -
            g -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Woverloaded-virtual -std=gnu++11 -MD -MT runtime/CMakeFiles/RealmRuntime.dir/realm/mem_impl.cc.o -MF CMakeFiles/RealmRuntime.dir/realm/mem_impl.cc.o.d -o CMakeFiles/RealmRuntime.dir/realm/mem_impl.cc
            .o -c /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-src/runtime/realm/mem_impl.cc
     189    [ 30%] Building CXX object runtime/CMakeFiles/RealmRuntime.dir/realm/metadata.cc.o
     190    cd /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime && /lustre/orion/csc439/scratch/wspear/crusher/23.02/spack/lib/spack/env/gcc/g++  -I/opt/rocm-5.4.0/include -I/opt/rocm-5.4.0/../include -I/t
            mp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-src/runtime -I/tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime -march=native -D__HIP_PLATFORM_HCC__ -O2 -
            g -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Woverloaded-virtual -std=gnu++11 -MD -MT runtime/CMakeFiles/RealmRuntime.dir/realm/metadata.cc.o -MF CMakeFiles/RealmRuntime.dir/realm/metadata.cc.o.d -o CMakeFiles/RealmRuntime.dir/realm/metadata.cc
            .o -c /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-src/runtime/realm/metadata.cc
     191    [ 31%] Building CXX object runtime/CMakeFiles/RealmRuntime.dir/realm/module.cc.o

     ...

     295    cd /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk/runtime && /lustre/orion/csc439/scratch/wspear/crusher/23.02/spack/opt/spack/linux-sles15-zen3/gcc-11.2.0/cmake-3.25.2-btskcfzaljlsbup6mlvd6zrdxbhxte
            qr/bin/cmake -E cmake_link_script CMakeFiles/RealmRuntime.dir/link.txt --verbose=1
     296    /usr/bin/ar qc ../lib/librealm.a CMakeFiles/RealmRuntime.dir/realm/transfer/lowlevel_disk.cc.o CMakeFiles/RealmRuntime.dir/realm/transfer/channel.cc.o CMakeFiles/RealmRuntime.dir/realm/transfer/channel_disk.cc.o CMakeFiles/RealmRuntime.dir/realm/transfer/t
            ransfer.cc.o CMakeFiles/RealmRuntime.dir/realm/transfer/lowlevel_dma.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/byfield.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/partitions.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/setops.cc.o CMakeFiles/Real
            mRuntime.dir/realm/deppart/sparsity_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/bgwork.cc.o CMakeFiles/RealmRuntime.dir/realm/event_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/faults.cc.o CMakeFiles/RealmRuntime.dir/realm/idx_impl.cc.o CMakeFiles/RealmRunt
            ime.dir/realm/inst_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/inst_layout.cc.o CMakeFiles/RealmRuntime.dir/realm/machine_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/mem_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/metadata.cc.o CMakeFiles/RealmRuntime.dir/
            realm/module.cc.o CMakeFiles/RealmRuntime.dir/realm/nodeset.cc.o CMakeFiles/RealmRuntime.dir/realm/numa/numa_module.cc.o CMakeFiles/RealmRuntime.dir/realm/numa/numasysif.cc.o CMakeFiles/RealmRuntime.dir/realm/operation.cc.o CMakeFiles/RealmRuntime.dir/real
            m/proc_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/procset/procset_module.cc.o CMakeFiles/RealmRuntime.dir/realm/rsrv_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/runtime_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/sampling_impl.cc.o CMakeFiles/RealmRuntime
            .dir/realm/subgraph_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/tasks.cc.o CMakeFiles/RealmRuntime.dir/realm/threads.cc.o CMakeFiles/RealmRuntime.dir/realm/hip/hip_module.cc.o CMakeFiles/RealmRuntime.dir/realm/hip/hip_internal.cc.o CMakeFiles/RealmRuntime.
            dir/realm/activemsg.cc.o CMakeFiles/RealmRuntime.dir/realm/network.cc.o CMakeFiles/RealmRuntime.dir/realm/cmdline.cc.o CMakeFiles/RealmRuntime.dir/realm/codedesc.cc.o CMakeFiles/RealmRuntime.dir/realm/logging.cc.o CMakeFiles/RealmRuntime.dir/realm/mutex.cc
            .o CMakeFiles/RealmRuntime.dir/realm/profiling.cc.o CMakeFiles/RealmRuntime.dir/realm/timers.cc.o CMakeFiles/RealmRuntime.dir/realm/utils.cc.o CMakeFiles/RealmRuntime.dir/image_1_1.cc.o CMakeFiles/RealmRuntime.dir/preimage_1_1.cc.o CMakeFiles/RealmRuntime.
            dir/byfield_1_1.cc.o CMakeFiles/RealmRuntime.dir/image_1_2.cc.o CMakeFiles/RealmRuntime.dir/preimage_1_2.cc.o CMakeFiles/RealmRuntime.dir/byfield_1_2.cc.o CMakeFiles/RealmRuntime.dir/image_1_3.cc.o CMakeFiles/RealmRuntime.dir/preimage_1_3.cc.o CMakeFiles/R
            ealmRuntime.dir/byfield_1_3.cc.o CMakeFiles/RealmRuntime.dir/image_2_1.cc.o CMakeFiles/RealmRuntime.dir/preimage_2_1.cc.o CMakeFiles/RealmRuntime.dir/byfield_2_1.cc.o CMakeFiles/RealmRuntime.dir/image_2_2.cc.o CMakeFiles/RealmRuntime.dir/preimage_2_2.cc.o 
            CMakeFiles/RealmRuntime.dir/byfield_2_2.cc.o CMakeFiles/RealmRuntime.dir/image_2_3.cc.o CMakeFiles/RealmRuntime.dir/preimage_2_3.cc.o CMakeFiles/RealmRuntime.dir/byfield_2_3.cc.o CMakeFiles/RealmRuntime.dir/image_3_1.cc.o CMakeFiles/RealmRuntime.dir/preima
            ge_3_1.cc.o CMakeFiles/RealmRuntime.dir/byfield_3_1.cc.o CMakeFiles/RealmRuntime.dir/image_3_2.cc.o CMakeFiles/RealmRuntime.dir/preimage_3_2.cc.o CMakeFiles/RealmRuntime.dir/byfield_3_2.cc.o CMakeFiles/RealmRuntime.dir/image_3_3.cc.o CMakeFiles/RealmRuntim
            e.dir/preimage_3_3.cc.o CMakeFiles/RealmRuntime.dir/byfield_3_3.cc.o
     297    /usr/bin/ranlib ../lib/librealm.a
     298    make[2]: Leaving directory '/tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk'
     299    [ 64%] Built target RealmRuntime
     300    make[1]: Leaving directory '/tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-wi3oukk'
  >> 301    make: *** [Makefile:139: all] Error 2

See build log for details:
  /tmp/wspear/spack-stage/spack-stage-legion-21.03.0-wi3oukkelse3wob6iiqkji36rfveygup/spack-build-out.txt
```

Building legion @master instead gives us this error:

```
     96     -- Generating dependency file: /tmp/wspear/spack-stage/spack-stage-legion-master-xtrockw3n3xeyt7vlpdaz6iygxxxjuto/spack-build-xtrockw/runtime/CMakeFiles/hip_compile.dir/legion/hip_compile_generated_legion_redop.cu.o.depend.pre
     97     /opt/rocm-5.4.0/bin/hipcc -M /tmp/wspear/spack-stage/spack-stage-legion-master-xtrockw3n3xeyt7vlpdaz6iygxxxjuto/spack-src/runtime/legion/legion_redop.cu -o /tmp/wspear/spack-stage/spack-stage-legion-master-xtrockw3n3xeyt7vlpdaz6iygxxxjuto/spack-build-xtroc
            kw/runtime/CMakeFiles/hip_compile.dir/legion/hip_compile_generated_legion_redop.cu.o.depend.pre -fno-strict-aliasing --offload-target=gfx90a -I /tmp/wspear/spack-stage/spack-stage-legion-master-xtrockw3n3xeyt7vlpdaz6iygxxxjuto/spack-src/runtime -I /tmp/wsp
            ear/spack-stage/spack-stage-legion-master-xtrockw3n3xeyt7vlpdaz6iygxxxjuto/spack-src/runtime/legion -I /tmp/wspear/spack-stage/spack-stage-legion-master-xtrockw3n3xeyt7vlpdaz6iygxxxjuto/spack-build-xtrockw/runtime -fPIC -g -O2
  >> 98     clang-15: error: unsupported option '--offload-target=gfx90a'
  >> 99     CMake Error at hip_compile_generated_legion_redop.cu.o.cmake:146 (message):
     100      Error generating
     101      /tmp/wspear/spack-stage/spack-stage-legion-master-xtrockw3n3xeyt7vlpdaz6iygxxxjuto/spack-build-xtrockw/runtime/CMakeFiles/hip_compile.dir/legion/./hip_compile_generated_legion_redop.cu.o
```

With the latest version of the package the new default version of legion, 23.03.0, still fails to build.

```
==> Installing legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu
==> No binary for legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu found: installing from source
==> Using cached archive: /tmp/wspear/spack-devel/var/spack/cache/_source-cache/git//StanfordLegion/legion.git/legion-23.03.0.tar.gz
==> Warning: Fetching from mirror without a checksum!
  This package is normally checked out from a version control system, but it has been archived on a spack mirror.  This means we cannot know a checksum for the tarball in advance. Be sure that your connection to this mirror is secure!
==> No patches needed for legion
==> legion: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/tmp/wspear/spack-devel/opt/spack/linux-sles15-zen3/gcc-11.2.0/cmake-3.26.3-mkgekkiejybgobu5fhfqcozagtrbpd5g/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/tmp/wspear/spack-devel/opt/spack/linux-sles15-zen3/gcc-11.2.0/legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/tmp/wspear/spack-devel/opt/spack/linux-sles15-zen3/gcc-11.2.0/legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/lib;/tmp/wspear/spack-devel/opt/spack/linux-sles15-zen3/gcc-11.2.0/legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/lib64;/opt/rocm-5.3.0/lib;/opt/rocm-5.3.0/llvm/lib;/tmp/wspear/spack-devel/opt/spack/linux-sles15-zen3/gcc-11.2.0/zlib-1.2.13-d35p7mxmwabhpoqwah7g3vlgawooct7s/lib;/tmp/wspear/spack-devel/opt/spack/linux-sles15-zen3/gcc-11.2.0/kokkos-4.0.00-whx55y5dshqm6qf2dhxffg2yzjxrdt7o/lib64;/opt/cray/pe/gcc/11.2.0/snos' '-DCMAKE_PREFIX_PATH:STRING=/tmp/wspear/spack-devel/opt/spack/linux-sles15-zen3/gcc-11.2.0/kokkos-4.0.00-whx55y5dshqm6qf2dhxffg2yzjxrdt7o;/tmp/wspear/spack-devel/opt/spack/linux-sles15-zen3/gcc-11.2.0/gmake-4.4.1-6b72ibntxvoxfjc3vs32fsm3wlxccjgu;/tmp/wspear/spack-devel/opt/spack/linux-sles15-zen3/gcc-11.2.0/cmake-3.26.3-mkgekkiejybgobu5fhfqcozagtrbpd5g;/tmp/wspear/spack-devel/opt/spack/linux-sles15-zen3/gcc-11.2.0/zlib-1.2.13-d35p7mxmwabhpoqwah7g3vlgawooct7s;/opt/rocm-5.3.0/llvm;/opt/rocm-5.3.0/;/opt/rocm-5.3.0' '-DCMAKE_CXX_STANDARD:STRING=11' '-DLegion_EMBED_GASNet=OFF' '-DBUILD_SHARED_LIBS=OFF' '-DLegion_USE_HIP=ON' '-DLegion_GPU_REDUCTIONS=ON' '-DLegion_HIP_TARGET:STRING=ROCM' '-DLegion_HIP_ARCH:STRING=gfx90a' '-DLegion_HIJACK_HIP:BOOL=OFF' '-DLegion_USE_LIBDL=ON' '-DLegion_USE_ZLIB=ON' '-DLegion_MAX_DIM=3' '-DLegion_MAX_FIELDS=512' '-DBUILD_MARCH:STRING=' '/tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-src'

2 errors found in build log:
     21    -- Performing Test COMPILER_SUPPORTS_MABI_ALTIVEC
     22    -- Performing Test COMPILER_SUPPORTS_MABI_ALTIVEC - Failed
     23    -- Performing Test COMPILER_SUPPORTS_MVSX
     24    -- Performing Test COMPILER_SUPPORTS_MVSX - Failed
     25    -- Performing Test COMPILER_SUPPORTS_DEFCHECK
     26    -- Performing Test COMPILER_SUPPORTS_DEFCHECK - Failed
  >> 27    CMake Error at cmake/FindHIP.cmake:25 (include):
     28      include could not find requested file:
     29    
     30        /opt/rocm-5.3.0/cmake/FindHIP.cmake
     31    Call Stack (most recent call first):
     32      CMakeLists.txt:503 (find_package)
     33    

     ...

     39    -- Looking for pthread_create in pthreads - not found
     40    -- Looking for pthread_create in pthread
     41    -- Looking for pthread_create in pthread - found
     42    -- Found Threads: TRUE
     43    -- Performing Test COMPILER_SUPORTS_OVERLOADED_VIRTUAL_WARNING
     44    -- Performing Test COMPILER_SUPORTS_OVERLOADED_VIRTUAL_WARNING - Success
  >> 45    CMake Error at cmake/FindHIP.cmake:40 (HIP_GET_SOURCES_AND_OPTIONS):
     46      Unknown CMake command "HIP_GET_SOURCES_AND_OPTIONS".
     47    Call Stack (most recent call first):
     48      cmake/FindHIP.cmake:54 (hip_compile_base)
     49      runtime/CMakeLists.txt:921 (hip_compile)
     50    
     51    

See build log for details:
  /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-out.txt

```

As mentioned before, there seems to be more than one issue with this package. Manually adjusting the path to HIP in the package allows cmake to complete but the build then fails like this with the latest default version as well.

```
==> Installing legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu
==> No binary for legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu found: installing from source
==> Using cached archive: /tmp/wspear/spack-devel/var/spack/cache/_source-cache/git//StanfordLegion/legion.git/legion-23.03.0.tar.gz
==> Warning: Fetching from mirror without a checksum!
  This package is normally checked out from a version control system, but it has been archived on a spack mirror.  This means we cannot know a checksum for the tarball in advance. Be sure that your connection to this mirror is secure!
==> No patches needed for legion
==> legion: Executing phase: 'cmake'
==> legion: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    '/tmp/wspear/spack-devel/opt/spack/linux-sles15-zen3/gcc-11.2.0/gmake-4.4.1-6b72ibntxvoxfjc3vs32fsm3wlxccjgu/bin/make' '-j16'

5 errors found in build log:
     92     cd /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd/runtime && /tmp/wspear/spack-de
            vel/lib/spack/env/gcc/g++ -D__HIP_PLATFORM_AMD__ -I/opt/rocm-5.3.0/include -I/opt/rocm-5.3.0/../include -I/tmp/wspear/spack-stage/spack-st
            age-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-src/runtime -I/tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5
            wimrviayc7gwveyhu/spack-build-haguupd/runtime -O2 -std=gnu++11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Woverloaded-virtual 
            -MD -MT runtime/CMakeFiles/RealmRuntime.dir/realm/idx_impl.cc.o -MF CMakeFiles/RealmRuntime.dir/realm/idx_impl.cc.o.d -o CMakeFiles/RealmR
            untime.dir/realm/idx_impl.cc.o -c /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-src/runtime/re
            alm/idx_impl.cc
     93     cd /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd/runtime && /tmp/wspear/spack-de
            vel/lib/spack/env/gcc/g++ -D__HIP_PLATFORM_AMD__ -I/opt/rocm-5.3.0/include -I/opt/rocm-5.3.0/../include -I/tmp/wspear/spack-stage/spack-st
            age-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-src/runtime -I/tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5
            wimrviayc7gwveyhu/spack-build-haguupd/runtime -O2 -std=gnu++11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Woverloaded-virtual 
            -MD -MT runtime/CMakeFiles/RealmRuntime.dir/realm/inst_impl.cc.o -MF CMakeFiles/RealmRuntime.dir/realm/inst_impl.cc.o.d -o CMakeFiles/Real
            mRuntime.dir/realm/inst_impl.cc.o -c /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-src/runtime
            /realm/inst_impl.cc
     94     -- Removing /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd/runtime/CMakeFiles/hip
            _compile.dir/legion/./hip_compile_generated_legion_redop.cu.o
     95     /tmp/wspear/spack-devel/opt/spack/linux-sles15-zen3/gcc-11.2.0/cmake-3.26.3-mkgekkiejybgobu5fhfqcozagtrbpd5g/bin/cmake -E remove /tmp/wspe
            ar/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd/runtime/CMakeFiles/hip_compile.dir/legion/.
            /hip_compile_generated_legion_redop.cu.o
     96     -- Generating dependency file: /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd/run
            time/CMakeFiles/hip_compile.dir/legion/hip_compile_generated_legion_redop.cu.o.depend.pre
     97     /opt/rocm-5.3.0/bin/hipcc -M /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-src/runtime/legion/
            legion_redop.cu -o /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd/runtime/CMakeFi
            les/hip_compile.dir/legion/hip_compile_generated_legion_redop.cu.o.depend.pre -fno-strict-aliasing --offload-target=gfx90a -I /tmp/wspear/
            spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-src/runtime -I /tmp/wspear/spack-stage/spack-stage-legion-23
            .03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-src/runtime/legion -I /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrv
            iayc7gwveyhu/spack-build-haguupd/runtime -fPIC -O2
  >> 98     clang-15: error: unsupported option '--offload-target=gfx90a'
  >> 99     CMake Error at hip_compile_generated_legion_redop.cu.o.cmake:146 (message):
     100      Error generating
     101      /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd/runtime/CMakeFiles/hip_compile.d
            ir/legion/./hip_compile_generated_legion_redop.cu.o
     102    
     103    
  >> 104    make[2]: *** [runtime/CMakeFiles/hip_compile.dir/build.make:80: runtime/CMakeFiles/hip_compile.dir/legion/hip_compile_generated_legion_red
            op.cu.o] Error 1
     105    make[2]: Leaving directory '/tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd'
  >> 106    make[1]: *** [CMakeFiles/Makefile2:225: runtime/CMakeFiles/hip_compile.dir/all] Error 2
     107    make[1]: *** Waiting for unfinished jobs....
     108    [ 15%] Building CXX object runtime/CMakeFiles/RealmRuntime.dir/realm/inst_layout.cc.o
     109    cd /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd/runtime && /tmp/wspear/spack-de
            vel/lib/spack/env/gcc/g++ -D__HIP_PLATFORM_AMD__ -I/opt/rocm-5.3.0/include -I/opt/rocm-5.3.0/../include -I/tmp/wspear/spack-stage/spack-st
            age-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-src/runtime -I/tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5
            wimrviayc7gwveyhu/spack-build-haguupd/runtime -O2 -std=gnu++11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Woverloaded-virtual 
            -MD -MT runtime/CMakeFiles/RealmRuntime.dir/realm/inst_layout.cc.o -MF CMakeFiles/RealmRuntime.dir/realm/inst_layout.cc.o.d -o CMakeFiles/
            RealmRuntime.dir/realm/inst_layout.cc.o -c /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-src/r
            untime/realm/inst_layout.cc
     110    [ 15%] Building CXX object runtime/CMakeFiles/RealmRuntime.dir/realm/machine_impl.cc.o
     111    cd /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd/runtime && /tmp/wspear/spack-de
            vel/lib/spack/env/gcc/g++ -D__HIP_PLATFORM_AMD__ -I/opt/rocm-5.3.0/include -I/opt/rocm-5.3.0/../include -I/tmp/wspear/spack-stage/spack-st
            age-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-src/runtime -I/tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5
            wimrviayc7gwveyhu/spack-build-haguupd/runtime -O2 -std=gnu++11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Woverloaded-virtual 
            -MD -MT runtime/CMakeFiles/RealmRuntime.dir/realm/machine_impl.cc.o -MF CMakeFiles/RealmRuntime.dir/realm/machine_impl.cc.o.d -o CMakeFile
            s/RealmRuntime.dir/realm/machine_impl.cc.o -c /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-sr
            c/runtime/realm/machine_impl.cc
     112    [ 16%] Building CXX object runtime/CMakeFiles/RealmRuntime.dir/realm/mem_impl.cc.o

     ...

     224    cd /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd/runtime && /tmp/wspear/spack-de
            vel/opt/spack/linux-sles15-zen3/gcc-11.2.0/cmake-3.26.3-mkgekkiejybgobu5fhfqcozagtrbpd5g/bin/cmake -E cmake_link_script CMakeFiles/RealmRu
            ntime.dir/link.txt --verbose=1
     225    /usr/bin/ar qc ../lib/librealm.a CMakeFiles/RealmRuntime.dir/realm/transfer/lowlevel_disk.cc.o CMakeFiles/RealmRuntime.dir/realm/transfer/
            channel.cc.o CMakeFiles/RealmRuntime.dir/realm/transfer/channel_disk.cc.o CMakeFiles/RealmRuntime.dir/realm/transfer/transfer.cc.o CMakeFi
            les/RealmRuntime.dir/realm/transfer/lowlevel_dma.cc.o CMakeFiles/RealmRuntime.dir/realm/transfer/ib_memory.cc.o CMakeFiles/RealmRuntime.di
            r/realm/deppart/byfield.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/partitions.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/setops.cc
            .o CMakeFiles/RealmRuntime.dir/realm/deppart/sparsity_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/bgwork.cc.o CMakeFiles/RealmRuntime.dir/
            realm/event_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/faults.cc.o CMakeFiles/RealmRuntime.dir/realm/idx_impl.cc.o CMakeFiles/RealmRuntim
            e.dir/realm/inst_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/inst_layout.cc.o CMakeFiles/RealmRuntime.dir/realm/machine_impl.cc.o CMakeFil
            es/RealmRuntime.dir/realm/mem_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/metadata.cc.o CMakeFiles/RealmRuntime.dir/realm/module.cc.o CMak
            eFiles/RealmRuntime.dir/realm/nodeset.cc.o CMakeFiles/RealmRuntime.dir/realm/numa/numa_module.cc.o CMakeFiles/RealmRuntime.dir/realm/numa/
            numasysif.cc.o CMakeFiles/RealmRuntime.dir/realm/operation.cc.o CMakeFiles/RealmRuntime.dir/realm/proc_impl.cc.o CMakeFiles/RealmRuntime.d
            ir/realm/procset/procset_module.cc.o CMakeFiles/RealmRuntime.dir/realm/repl_heap.cc.o CMakeFiles/RealmRuntime.dir/realm/rsrv_impl.cc.o CMa
            keFiles/RealmRuntime.dir/realm/runtime_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/sampling_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/su
            bgraph_impl.cc.o CMakeFiles/RealmRuntime.dir/realm/tasks.cc.o CMakeFiles/RealmRuntime.dir/realm/threads.cc.o CMakeFiles/RealmRuntime.dir/r
            ealm/hip/hip_module.cc.o CMakeFiles/RealmRuntime.dir/realm/hip/hip_internal.cc.o CMakeFiles/RealmRuntime.dir/realm/hip/hip_access.cc.o CMa
            keFiles/RealmRuntime.dir/realm/activemsg.cc.o CMakeFiles/RealmRuntime.dir/realm/network.cc.o CMakeFiles/RealmRuntime.dir/realm/cmdline.cc.
            o CMakeFiles/RealmRuntime.dir/realm/codedesc.cc.o CMakeFiles/RealmRuntime.dir/realm/logging.cc.o CMakeFiles/RealmRuntime.dir/realm/mutex.c
            c.o CMakeFiles/RealmRuntime.dir/realm/profiling.cc.o CMakeFiles/RealmRuntime.dir/realm/timers.cc.o CMakeFiles/RealmRuntime.dir/realm/utils
            .cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/image_1_1.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/preimage_1_1.cc.o CMakeFiles/Real
            mRuntime.dir/realm/deppart/byfield_1_1.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/image_1_2.cc.o CMakeFiles/RealmRuntime.dir/realm/dep
            part/preimage_1_2.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/byfield_1_2.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/image_1_3.cc.o
             CMakeFiles/RealmRuntime.dir/realm/deppart/preimage_1_3.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/byfield_1_3.cc.o CMakeFiles/RealmRu
            ntime.dir/realm/deppart/image_2_1.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/preimage_2_1.cc.o CMakeFiles/RealmRuntime.dir/realm/deppa
            rt/byfield_2_1.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/image_2_2.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/preimage_2_2.cc.o C
            MakeFiles/RealmRuntime.dir/realm/deppart/byfield_2_2.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/image_2_3.cc.o CMakeFiles/RealmRuntime
            .dir/realm/deppart/preimage_2_3.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/byfield_2_3.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/
            image_3_1.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/preimage_3_1.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/byfield_3_1.cc.o CMak
            eFiles/RealmRuntime.dir/realm/deppart/image_3_2.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/preimage_3_2.cc.o CMakeFiles/RealmRuntime.d
            ir/realm/deppart/byfield_3_2.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/image_3_3.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/preim
            age_3_3.cc.o CMakeFiles/RealmRuntime.dir/realm/deppart/byfield_3_3.cc.o
     226    /usr/bin/ranlib ../lib/librealm.a
     227    make[2]: Leaving directory '/tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd'
     228    [ 63%] Built target RealmRuntime
     229    make[1]: Leaving directory '/tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd'
  >> 230    make: *** [Makefile:139: all] Error 2

See build log for details:
  /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-out.txt
```

I'm happy to provide additional logs or make adjustments to the package or install instructions to help track down a fix for this.

> I'm not sure I see the actual error above?

The spack.yaml included with original post shows how the host compiler and hip are set up:
```
spack:
  ...
  compilers:
  - compiler:
      spec: gcc@11.2.0
      paths:
        cc: gcc
        cxx: g++
        f77: gfortran
        fc: gfortran
      flags: {}
      operating_system: sles15
      target: any
      modules:
      - PrgEnv-gnu
      - gcc/11.2.0
      environment:
        unset:
        - XALT_OLCF
        - XL_LINKER
        - XALT_ETC_DIR
        remove_path:
          PATH: /sw/crusher/xalt/1.3.0/bin
          PYTHONPATH: /sw/crusher/xalt/1.3.0/site
          LD_LIBRARY_PATH: /sw/aaims/crusher/build/smartbench/lib
      extra_rpaths: []
  ...
  packages:
    ...
    hip:
      buildable: false
      externals:
      - spec: hip@5.3.0
        prefix: /opt/rocm-5.3.0
        modules: [rocm/5.3.0]
        extra_attributes:
          compilers:
            c: /opt/rocm-5.3.0/llvm/bin/clang++
            c++: /opt/rocm-5.3.0/llvm/bin/clang++
            hip: /opt/rocm-5.3.0/hip/bin/hipcc
        environment:
          set:
            MPICH_GPU_SUPPORT_ENABLED: 1
  ...
```

The problems are two-fold:

1. With `legion`'s current package.py, it cannot find HIP properly:
```
  >> 27    CMake Error at cmake/FindHIP.cmake:25 (include):
     28      include could not find requested file:
     29    
     30        /opt/rocm-5.3.0/cmake/FindHIP.cmake
     31    Call Stack (most recent call first):
     32      CMakeLists.txt:317 (find_package)
```

2. Even if the package.py is tweaked so that CMake can find HIP, `hipcc` does not get used where it should:
(excerpt from @wspear's last post)
```
  >> 98     clang-15: error: unsupported option '--offload-target=gfx90a'
  >> 99     CMake Error at hip_compile_generated_legion_redop.cu.o.cmake:146 (message):
     100      Error generating
     101      /tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd/runtime/CMakeFiles/hip_compile.d
            ir/legion/./hip_compile_generated_legion_redop.cu.o
     102    
     103    
  >> 104    make[2]: *** [runtime/CMakeFiles/hip_compile.dir/build.make:80: runtime/CMakeFiles/hip_compile.dir/legion/hip_compile_generated_legion_red
            op.cu.o] Error 1
     105    make[2]: Leaving directory '/tmp/wspear/spack-stage/spack-stage-legion-23.03.0-haguupdeij263g5wimrviayc7gwveyhu/spack-build-haguupd'
  >> 106    make[1]: *** [CMakeFiles/Makefile2:225: runtime/CMakeFiles/hip_compile.dir/all] Error 2
     107    make[1]: *** Waiting for unfinished jobs....
```

Notice:
```
  >> 98     clang-15: error: unsupported option '--offload-target=gfx90a'
```

We usually see this last type of error when the wrong compiler is being used to compile code that needs `hipcc`.

A number of other packages deal with this by setting `CMAKE_CXX_COMPILER` to `hipcc` for their `+rocm` variant:
* [kokkos +rocm](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/kokkos/package.py#L327-L328)
* [raja +rocm](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/raja/package.py#L113-L114)
* [hpx +rocm](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/hpx/package.py#L239-L240)
* [strumpack +rocm](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/strumpack/package.py#L151-L152)

> Where in your logs can I see the CMake configure line Legion was built with?

From `spack-build-out.txt` (long paths removed for clarity):
```
cmake \
 -G "Unix Makefiles" \
 -DCMAKE_INSTALL_PREFIX:STRING=... \
 -DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo \
 -DBUILD_TESTING:BOOL=OFF \
 -DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF \
 -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
 -DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON \
 -DCMAKE_INSTALL_RPATH:STRING=... \
 -DCMAKE_PREFIX_PATH:STRING="...;/opt/rocm-5.3.0/llvm;/opt/rocm-5.3.0/" \
 -DCMAKE_CXX_STANDARD:STRING=11 \
 -DLegion_EMBED_GASNet=OFF \
 -DBUILD_SHARED_LIBS=OFF \
 -DLegion_USE_HIP=ON \
 -DLegion_GPU_REDUCTIONS=ON \
 -DLegion_HIP_TARGET:STRING=ROCM \
 -DLegion_HIP_ARCH:STRING=gfx90a \
 -DLegion_HIJACK_HIP:BOOL=OFF \
 -DLegion_USE_LIBDL=ON \
 -DLegion_USE_ZLIB=ON \
 -DLegion_MAX_DIM=3 \
 -DLegion_MAX_FIELDS=512 \
 ...
```

I think the issue here is that `FindHIP.cmake` has been moved around between different ROCm versions (along with other changes to the location of various ROCm components).

From the original CMake error, it is looking for `FindHIP.cmake` here:
```
  >> 27    CMake Error at cmake/FindHIP.cmake:25 (include):
     28      include could not find requested file:
     29    
     30        /opt/rocm-5.3.0/cmake/FindHIP.cmake
```

But that is not where it lives w/ ROCm 5.3.0. Instead, it is here:
```
/opt/rocm-5.3.0/lib/cmake/hip/FindHIP.cmake
```

Worth noting in `spack-build-env.txt` is that `HIP_PATH` is being set:
```
...
HIP_PATH=/opt/rocm-5.3.0; export HIP_PATH
...
```

> Is the following patch to Legion sufficient to make it build?
> ...

I manually applied the patch and it did get us past the initial CMake error, but now we are seeing the error Wyatt reported above, and that I described as #2 here (https://github.com/spack/spack/issues/37232#issuecomment-1550533129)

```
==> Installing legion-23.03.0-li53u7buzh3mmqxt4dppcaqgrap46f2u
==> No binary for legion-23.03.0-li53u7buzh3mmqxt4dppcaqgrap46f2u found: installing from source
==> Applied patch /mnt/bb/lpeyrala/PrgEnv-gnu/spack/var/spack/repos/builtin/packages/legion/findhip.patch
==> legion: Executing phase: 'cmake'
==> legion: Executing phase: 'build'
...
  >> 130    clang-15: error: unsupported option '--offload-target=gfx90a'
  >> 131    CMake Error at hip_compile_generated_legion_redop.cu.o.cmake:146 (message):
     132      Error generating
     133      /tmp/lpeyrala/spack-stage/spack-stage-legion-23.03.0-li53u7buzh3mmqxt4dppcaqgrap46f2u/spack-build-li53u7b/runtime/CMakeFiles/hip_compile.dir/legion/./hip_compile_
            generated_legion_redop.cu.o
     134
     135
  >> 136    make[2]: *** [runtime/CMakeFiles/hip_compile.dir/build.make:80: runtime/CMakeFiles/hip_compile.dir/legion/hip_compile_generated_legion_redop.cu.o] Error 1
     137    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-legion-23.03.0-li53u7buzh3mmqxt4dppcaqgrap46f2u/spack-build-li53u7b'
  >> 138    make[1]: *** [CMakeFiles/Makefile2:225: runtime/CMakeFiles/hip_compile.dir/all] Error 2
     139    make[1]: *** Waiting for unfinished jobs....
...
```

It looks like some code needs `hipcc` but is not getting it:
```
  >> 130    clang-15: error: unsupported option '--offload-target=gfx90a'
```

--- a/var/spack/repos/builtin/packages/legion/hip-offload-arch.patch
+++ b/var/spack/repos/builtin/packages/legion/hip-offload-arch.patch
@@ -1,0 +1,24 @@
+diff -ruN spack-src/CMakeLists.txt spack-src-patched/CMakeLists.txt
+--- spack-src/CMakeLists.txt	2023-05-18 14:18:00.897162488 -0400
++++ spack-src-patched/CMakeLists.txt	2023-05-18 14:20:09.532413649 -0400
+@@ -495,7 +495,7 @@
+     if("${Legion_HIP_ARCH}" STREQUAL "")
+       set(HIP_GENCODE "")
+     else()
+-      set(HIP_GENCODE "--offload-target=${Legion_HIP_ARCH}")
++      set(HIP_GENCODE "--offload-arch=${Legion_HIP_ARCH}")
+     endif()
+   endif()
+ 
+diff -ruN spack-src/runtime/runtime.mk spack-src-patched/runtime/runtime.mk
+--- spack-src/runtime/runtime.mk	2023-05-18 14:18:00.969164248 -0400
++++ spack-src-patched/runtime/runtime.mk	2023-05-18 14:20:51.317470176 -0400
+@@ -499,7 +499,7 @@
+       HIPCC_FLAGS	+= -O2
+     endif
+     ifneq ($(strip $(HIP_ARCH)),)
+-      HIPCC_FLAGS	+= --offload-target=$(HIP_ARCH)
++      HIPCC_FLAGS	+= --offload-arch=$(HIP_ARCH)
+     endif
+     LEGION_LD_FLAGS	+= -lm -L$(HIP_PATH)/lib -lamdhip64
+   else ifeq ($(strip $(HIP_TARGET)),CUDA)

--- a/var/spack/repos/builtin/packages/legion/package.py
+++ b/var/spack/repos/builtin/packages/legion/package.py
@@ -70,6 +70,9 @@ class Legion(CMakePackage, ROCmPackage):
     depends_on("kokkos@3.3.01:~cuda", when="+kokkos~cuda")
     depends_on("kokkos@3.3.01:~cuda+openmp", when="+kokkos+openmp")
 
+    # https://github.com/spack/spack/issues/37232#issuecomment-1553376552
+    patch("hip-offload-arch.patch", when="+rocm")
+
     # HIP specific
     variant(
         "hip_hijack",

--- a/var/spack/repos/builtin/packages/legion/package.py
+++ b/var/spack/repos/builtin/packages/legion/package.py
@@ -71,7 +71,7 @@ class Legion(CMakePackage, ROCmPackage):
     depends_on("kokkos@3.3.01:~cuda+openmp", when="+kokkos+openmp")
 
     # https://github.com/spack/spack/issues/37232#issuecomment-1553376552
-    patch("hip-offload-arch.patch", when="+rocm")
+    patch("hip-offload-arch.patch", when="@23.03.0 +rocm")
 
     # HIP specific
     variant(

